### PR TITLE
fix(Authentication): ESSNTL-5018 - Show "Not Authorised" with no permissions

### PR DESCRIPTION
**Pull request by bastilian**
# Description

Associated Jira ticket: #ESSNTL-5018 (issue)

This lifts up the RBAC check in sed-frontend in order to avoid querying the API with an account that doesn't have (any) permissions.

# How to test the PR

1) Create an account with no roles/permissions assigned (related to RHC)
2) Verify RHC shows the "Not authorised page"

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Review by AsToNlele — COMMENTED** [review]
This works great, with no roles I see the "You do not have access...".

When I add the RHC viewer role and have no other roles, the page loads as expected, but I get this error:
image

When I add Inventory Hosts Read, I get no error notification.

Not sure if this is expected, but this might be out of the scope for this ticket.
What do you think?

**Comment by bastilian** [comment]
@AsToNlele Thanks for testing and reviewing! The `Authentication` component is now lifted up to the `AppEntry` component and should now prevent any API calls.

File: src/AppEntry.js [hunk 1]
```diff
@@ -13,7 +14,9 @@ const AppEntry = () => {
       }}
     >
       <Provider store={registry.getStore()}>
-        <App />
+        <Authentication>
```
Comment (gkarat) [inline src/AppEntry.js]: This should be wrapped in QueryClientProvider. Right now, if you open /insights/connector, you get
```
No QueryClient set, use QueryClientProvider to set one
```

**Review by gkarat — CHANGES_REQUESTED** [review]

File: src/Routes.js [hunk 2]
```diff
@@ -31,36 +30,34 @@ const AppRoutes = () => {
...
+      <Route
+        path={paths.activationKeys}
+        element={
+          <SuspenseWrapped>
+            <ActivationKeys />
+          </SuspenseWrapped>
+        }
+      />
+      <Route
+        path={paths.connector}
```
Comment (gkarat) [inline src/Routes.js:53]: I am not sure how exactly this PR is causing it, but if I navigate by clicking on "View log" in the upper top corner, I get an error
```
(0 , react_router_dom__WEBPACK_IMPORTED_MODULE_4__.useHistory) is not a function
```

Comment (gkarat) [reply src/Routes.js:53]: it's not reproducible on stage c.r.c. (master version)

**Review by gkarat — CHANGES_REQUESTED** [review]

**Review by gkarat — APPROVED** [review]
LGTM! the previous issues seem to be resolved, and it worked nicely locally... thanks @bastilian

**Pull request merged**